pandas/io/parsers.py

```diff
@@ -1419,6 +1420,56 @@ def __init__(self, kwds):
         # keep references to file handles opened by the parser itself
         self.handles = []
 
+    def _validate_parse_dates_presence(self, columns: List[str]) -> None:
+        """
+        Check if parse_dates are in columns.
+
+        If user has provided names for parse_dates, check if those columns
+        are available.
+
+        Parameters
+        ----------
+        columns : list
+            list of names of the dataframe.
+
+        Raises
+        ------
+        ValueError
+            If column to parse_date is not in dataframe.
+
+        """
+        if isinstance(self.parse_dates, list):
```

Comment:
IIUC can just use `itertools.chain.from_iterable` for this instead of rolling our own impl

Reply:
If I were to replace the current flow with `from_iterable`, I'll need to do 

```
-            cols_needed = []
-            for col in self.parse_dates:
-                if isinstance(col, list):
-                    cols_needed.extend(col)
-                else:
-                    cols_needed.append(col)
+            cols_needed = chain.from_iterable([col if isinstance(col, list) else [col] for col in self.parse_dates ])
         elif isinstance(self.parse_dates, dict):
```

I need these check to handle the case when `parse_dates=[["date", "time"], "date2" ]`. Here `parse_dates` is a list whose first element is still a list and second is a string, and I need to all the columns refererenced `date, time, date2`.

Reply:
I have replaced `extend` and `append`, let me know if it is better.